public/js/components/card-form.jsx

```diff
@@ -123,11 +138,16 @@ export default class CardForm extends Component {
   handleSubmit = (e) => {
     e.preventDefault();
     this.setState({isSubmitting: true});
-    this.props.handleCardSubmit({
+    var formData = {
       number: this.state.card,
       cvv: this.state.cvv,
       expiration: this.state.expiration,
-    });
+    };
+    if (this.props.emailFieldRequired) {
+      this.props.handleCardSubmit(formData, this.state.email);
```

Comment:
since email is optional I think it would be better to pass it in as a keyword:
````
handleCardSubmit(..., {email: this.state.email});
````